client/jetpack-cloud/sections/sidebar-navigation/lib/sidebar.ts

```diff
@@ -1,14 +1,30 @@
 import page from 'page';
 import { itemLinkMatches } from 'calypso/my-sites/sidebar/utils';
+import {
+	JETPACK_MANAGE_ASSIGN_LICENCE_LINK,
+	JETPACK_MANAGE_ISSUE_LICENCE_LINK,
+	JETPACK_MANAGE_LICENCES_LINK,
+} from './constants';
 
 /**
  * Determines whether a menu item should be selected based on the current URL path.
  * @param link The link to compare against the current URL path.
+ * @param path The current URL path.
  * @returns A boolean value indicating whether the link matches the current URL path.
  */
-export const isMenuItemSelected = ( link: string ) => {
-	const pathname = window.location.pathname;
-	return itemLinkMatches( link, pathname );
+export const isMenuItemSelected = ( link: string, path: string ) => {
+	let overridenPath = path;
+
+	// For Assign and Issue license path, we will override it to be license path to make the menu selected.
+	if (
+		( link === JETPACK_MANAGE_LICENCES_LINK &&
+			path.startsWith( JETPACK_MANAGE_ASSIGN_LICENCE_LINK ) ) ||
+		path.startsWith( JETPACK_MANAGE_ISSUE_LICENCE_LINK )
+	) {
+		overridenPath = JETPACK_MANAGE_LICENCES_LINK;
+	}
```

Comment:
What do you think about moving this logic into itemLinkMatches? As you can see there we have similar logic, e.g.:

```
	// For `/store/stats/*` and `/google-my-business/stats/*` paths, show Stats menu as selected.
	if (
		currentPath.startsWith( '/store/stats/' ) ||
		currentPath.startsWith( '/google-my-business/stats/' )
	) {
		return path.startsWith( '/stats/' );
	}
``` 

I think it's better to have everything in one place for consistency.

Reply:
I had second thoughts about putting this logic in the `itemLinkMatches` because it felt like it was just a workaround for the current sidebar menu. However, upon further reflection, I realized that the `itemLinkMatches` helper function is only used for the sidebar and nothing else, so I agree that it would be better to move it there.

Thanks for the suggestion 👍